src/CoreWCF.Primitives/src/CoreWCF/ServiceHostObjectModel.cs

```diff
@@ -96,6 +96,7 @@ protected override ServiceDescription CreateDescription(out IDictionary<string,
 
             ReflectedContractCollection reflectedContracts = new ReflectedContractCollection();
             List<Type> interfaces = ServiceReflector.GetInterfaces<TService>();
+            TypeLoader<TService>.ServiceProvider = _serviceProvider;
             for (int i = 0; i < interfaces.Count; i++)
```

Comment:
This is fragile and can cause problems in a rare scenario. With a single WebHost being built, this is fine, but if you had 2 different threads creating 2 different WebHost instances for the same service type at the same time, you could get the service provider leak between WebHost build pipelines and end up with some really difficult to diagnose bugs.

Reply:
I now store the `ServiceProvider` in `ContractDescription`. That's not an issue to store it after object creation as it is used lazyly when first request comes in to populate the lazy AuthorizeData property. This allow us not to break the public API `ContractDescription.GetContract`. please see https://github.com/CoreWCF/CoreWCF/pull/850/commits/d9e9dac4bf5685df166bf9d85fceb3a4602a2a34

Reply:
I finally dropped this code